File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int ways[46] = { -1 };
    int climbStairs(int n) {
        if (ways[0] == -1) {
            ways[0] = ways[1] = 1;
            for (int i = 2; i <= 45; i++) {
                ways[i] = ways[i - 1] + ways[i - 2];
            }
        }
        return ways[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    
    public static int[] result;
    static {
        result = new int[46];
        result[0] = 1;
        result[1] = 1;
        for (int i = 2; i <= 45; i++) {
            result[i] = result[i - 1] + result[i - 2];
        }
    }
    
    public int climbStairs(int n) {
        return result[n];
    }
}